src/test/java/com/glencoesoftware/omero/zarr/ZarrPixelBufferTest.java

```diff
@@ -991,89 +919,77 @@ public void testOrderXYZTC() throws IOException, InvalidRangeException {
+                            plane, zpbuf.getPixelsType(), false);
+                        int planeNumber = FormatTools.getIndex(
+                            DimensionOrder.VALUE_XYZCT,
+                            sizeZ, sizeC, sizeT, sizeZ * sizeC * sizeT,
+                            z, c, t);
+                        Assert.assertArrayEquals(
+                            Arrays.toString(seriesPlaneNumberZCT),
+                            new int[] {0, planeNumber, z, c, t},
+                            seriesPlaneNumberZCT);
+                    }
+                }
+            }
         }
     }
 
     private void testDimensions(int sizeX, int sizeY, int sizeZ, int sizeC, int sizeT)
             throws IOException, InvalidRangeException {
         
-        String order = DimensionOrder.VALUE_XYZCT;
-        if (sizeT == 0) {
-            order.replace("T", "");
-        }
-        if (sizeZ == 0) {
-            order.replace("Z", "");
-        }
-        if (sizeC == 0) {
-            order.replace("C", "");
-        }
-        
-        int textX = 10;
-        int textY = 10;
-        Path testZarrPath = tmpDir.getRoot().toPath().resolve("test.zarr");
-        TestZarr testZarr = new TestZarr()
-            .setPath(testZarrPath)
-            .setOverwrite(true)
-            .setSizeX(sizeX)
-            .setSizeY(sizeY)
-            .setSizeZ(sizeZ)
-            .setSizeT(sizeT)
-            .setSizeC(sizeC)
-            .setOrder(new StringBuilder(order).reverse().toString())
-            .setTextX(textX)
-            .setTextY(textY)
-            .init()
-            .createImage()
-            .createMetadata();
-
-        int pixZ = sizeZ > 0 ? sizeZ : 1;
-        int pixC = sizeC > 0 ? sizeC : 1;
-        int pixT = sizeT > 0 ? sizeT : 1;
-        
-        Pixels pixels = new Pixels(
-            null, new PixelsType(PixelsType.VALUE_INT32), 
-            sizeX, sizeY, pixZ, pixC, pixT, "", new DimensionOrder(DimensionOrder.VALUE_XYZCT));
+        Pixels pixels = new Pixels(null, null, sizeX, sizeY, sizeZ, sizeC, sizeT, "", null);
+        Path output = writeTestZarr(sizeT, sizeC, sizeZ, sizeY, sizeX, "uint16", "--compact");
         
-        int expectedTests = pixC * pixT * pixZ;
-        int testCount = 0;
-        try (ZarrPixelBuffer zpbuf = createPixelBuffer(
-                pixels, testZarrPath.resolve("0"), sizeX, sizeY)) {
-            for (int t = 0; t < Math.max(sizeT, 1); t++) {
-                for (int z = 0; z < Math.max(sizeZ, 1); z++) {
-                    for (int c = 0; c < Math.max(sizeC, 1); c++) {
-                        byte[] expected = testZarr.generateGreyscaleImageWithText(c, z, t);
-                        byte[] actual = zpbuf.getPlane(z, c, t).getData().array();
-                        Assert.assertArrayEquals(expected, actual);
-                        testCount++;
+        try (ZarrPixelBuffer zpbuf = createPixelBuffer(pixels, output.resolve("0"), sizeX, sizeY)) {
```

Comment:
Worth putting the special pixels checks in a separate method, since both `testDimensions` and `testOrder` do this?